cli/src/commands/upload.ts

```diff
@@ -75,22 +79,36 @@ export default class Upload extends BaseCommand {
         });
 
         let skipUpload = false;
-        if (!options.skipHash) {
+
+        let skipAsset = false;
+        let existingAssetId: string | undefined = undefined;
+
+        if (!options.skipHash && !options.album) {
           const assetBulkUploadCheckDto = { assets: [{ id: asset.path, checksum: await asset.hash() }] };
 
           const checkResponse = await this.immichApi.assetApi.checkBulkUpload({
             assetBulkUploadCheckDto,
           });
 
           skipUpload = checkResponse.data.results[0].action === 'reject';
+
+          const isDuplicate = checkResponse.data.results[0].reason === 'duplicate';
+          if (isDuplicate) {
+            existingAssetId = checkResponse.data.results[0].assetId;
+          }
+
+          skipAsset = skipUpload && !isDuplicate;
         }
 
-        if (!skipUpload) {
+        if (!skipAsset) {
           if (!options.dryRun) {
-            const formData = asset.getUploadFormData();
-            const res = await this.uploadAsset(formData);
+            if (!skipUpload) {
+              const formData = asset.getUploadFormData();
+              const res = await this.uploadAsset(formData);
+              existingAssetId = res.data.id;
+            }
 
-            if (options.album && asset.albumName) {
+            if (asset.albumName !== undefined) {
```

Comment:
Not sure about this check alone: `asset.albumName` is always defined to parent folder name when checked after `asset.process()`. 
`options.album` and `options.albumName` are not tested, it means the cli is always going to create and add the asset to an album. I think the expected behavior is to only add assets to an album if `-A` or `-a` is defined. 
I would use something like  `if ((options.album || options.albumName) && asset.albumName !== undefined) {`

Reply:
Thanks, added another test to catch this case and fixed the code